docs/audio.rst

```diff
@@ -215,6 +244,153 @@ Sound Effects Example
+        via the microphone, or played via the ``audio.play()`` function.
+
+    .. py:function:: set_rate(sample_rate)
+
+        Configure the sampling rate associated with the data in the
+        ``AudioRecording`` instance.
+
+        :param sample_rate: The sample rate to set.
+
+    .. py:function:: get_rate()
+
+        Return the configured sampling rate for this
+        ``AudioRecording`` instance.
+
+        :return: The configured sample rate.
+
+    .. py:function:: copy()
+
+        :returns: a copy of the ``AudioRecording``.
+
+    .. py:function:: track(start_ms=0, end_ms=-1)
+
+        Create an `AudioTrack <#audio.AudioTrack>`_ instance from a portion of
+        the data in this ``AudioRecording`` instance.
+
+        :param start_ms: Where to start of the track in milliseconds.
+        :param end_ms: The end of the track in milliseconds.
+            If the default value of ``-1`` is provided it will end the track
+            at the end of the AudioRecording.
+
+When an AudioRecording is used to record data from the microphone,
+increasing the sampling rate increases the sound quality,
+but it also increases the amount of memory used.
+
+During playback, increasing the sampling rate speeds up the sound
+and decreasing the sample rate slows it down.
+
+AudioTrack
+----------
+
+An ``AudioTrack`` can be created from an ``AudioRecording`` or ``bytearray``
+and individual bytes can be accessed and modified like elements in a list.
+
+This class is useful to modify the audio data in an ``AudioRecording`` or
+to create precisely sized audio data buffers for sending and receiving them
+via communication protocols like radio or serial.
+
+.. py:class::
+    AudioTrack(buffer, rate=7812)
+
+    The ``AudioTrack`` object points to the data provided by the input buffer,
+    which can be an ``AudioRecording``, or a buffer-like object like a
+    ``bytearray``.
+    It also contains its own sampling rate, so multiple ``AudioTrack``
+    instances pointing to the same buffer can have different rates and won't
+    affect the rate of the original buffer.
+
+    :param buffer: The buffer containing the audio data.
+    :param rate: The sampling rate at which data will be stored
+        via the microphone, or played via the ``audio.play()`` function.
+
+    .. py:function:: set_rate(sample_rate)
+
+        Configure the sampling rate associated with the data in the
+        ``AudioTrack`` instance.
+
+        :param sample_rate: The sample rate to set.
+
+    .. py:function:: get_rate()
+
+        Return the configured sampling rate for this
+        ``AudioTrack`` instance.
+
+        :return: The configured sample rate.
+
+    .. py:function:: copyfrom(other)
+
+        Overwrite the data in this ``AudioTrack`` with the data from another
+        ``AudioTrack``, ``AudioRecording`` or buffer-like object like
+        a ``bytes`` or ``bytearray`` instance.
+
+        :param other: Buffer-like instance from which to copy the data.
+
```

Comment:
Will need to document access via subscript and slicing, and `len(audio_track)` operator.

Reply:
Is that commonly documented via the dunder methods?
That might  be a bit hard to understand for some of our audience, maybe we should add an explanation before or after the class documentation, or the in the class description?

Reply:
Would be great to document these somehow as otherwise it's trial and error/reading the C.

I think for AudioTrack we have:

```
    @overload
    def __getitem__(self, i: int) -> int: ...
    @overload
    def __getitem__(self, s: slice) -> AudioTrack: ...
    def __setitem__(self, i: int, x: int) -> None: ...

    def __add__(self, v: AudioTrack) -> AudioTrack: ...
    def __iadd__(self, v: AudioTrack) -> AudioTrack: ...
    def __sub__(self, v: AudioTrack) -> AudioTrack: ...
    def __isub__(self, v: AudioTrack) -> AudioTrack: ...
    def __mul__(self, v: float) -> AudioTrack: ...
    def __imul__(self, v: float) -> AudioTrack: ...
 ```

So slice access but not slice assignment.